### ISSUE TITLE
Advise a decelerated development speed

### ISSUE THREAD

**Issue by taiyi747** [xtekky/gpt4free#120]
First of all, I am happy to see that this project has reached 4.2k stars. However, I have also noticed some negative things.

On the positive side, I have seen many capable people contributing to this repository and adding many things to it. When I discovered this project, I did not expect it to have so many followers.

On the negative side, I have seen many people abusing this project by integrating it into chatbots, writing plugins, and publishing them on platforms like YouTube to gain traffic and money. Although these issues are relatively small, there is a bigger problem that has arisen.

Some issues raised by people, such as the one in this repository (https://github.com/gitzboy/testremote), are causing many free GPT websites to shut down rapidly. This is because these websites are being published on GitHub, and a large number of people are flooding in, causing the following problems:

1.Website downtime
2.OpenAI accounts being suspended
3.Websites being forced to charge fees
4.Website operators not having the time and energy to maintain these websites
5.Websites closing down due to lack of economic sources

90% of these websites are operated by individuals or small teams, and they do not have the economic resources to support such a large amount of traffic. Their original intention was to provide a long-term free service, but they are now receiving an overwhelming amount of traffic in a short period of time.

In my opinion, our reverse-engineering goals should be focused on large company websites, start-up company websites related to OpenAI (such as the previous "nat" website), developer community websites with economic support, and websites that have been providing stable and free GPT services for a long time.

We should not focus on small websites that individuals do not have the economic ability to support the traffic brought by this repository. The code will also quickly become obsolete, making us appear very blind.

In summary, I just want to express my opinion that we should slow down the development speed, improve the project, and save time to deal with the existing services if they fail.

Suggestions: Create a topic to confirm whether some websites are worth being our reverse-engineering targets. Prioritize improving the code in the repository.

My words may be a bit extreme, and I apologize in advance. I hope everyone will not be angry after reading this, and if you have any questions, please feel free to express them.

**Comment by xtekky** [comment]
Hi, if you look at this repo, it does mainly use big sites like poe.com, phind.com, ora.sh, you.com etc...
Due to the big guys getting scared and patching everything, too keep this project alive I did include some smaller gpt projects.

I would not consider myself the bad guy in this situation, a smaller site can always contact me and ask for a takedown of their api, and this repo is also there to rais awareness about general websecurity:

A smaller site could include:
- captcha
- cloudflare
- accounts system and ratelimiting

aswell as the one I suggest the most which will using crypto and hashing to secure the api. Some site implemented this but everything i saw till now is some completely open non obfuscated code with some md5 hash or JWT. I mean at this point its their fault.

I will be looking into this in the future as I noticed that every site I use gets patched shortly after, but this happened with sites i used just personally fo chatbot.sex for example. We will be creating a system to target sites that would not be highly affected by this repo.

If i slow down it would rather be because of personal reasons, if I busy etc..., but I will always fully commit to this project and stopping contributions is not an option at all.

I think a good solution would be that either site owners contact me or that we choose our tagets more carefully, even if at the moment i find the sites used reasonnable

**Comment by xtekky** [comment]
This is also not really this repo's fault, for instance i have been getting over the norms traffic on my own site, even if the api is very protected and is not used through automated requests

**Comment by taiyi747** [comment]
haha, I mainly thought of these things because of the "suggest websites to reverse" section. I have seen too many projects like this before, and the bigger the project, the more effort you have to put in and the greater the risks it faces. So far, I think you have done a great job.

**Comment by xtekky** [comment]
all these websites are just "suggestions" for now and not implemented, most of the sites are bad and do not fill in certain criteria so I won't reverse most of them

**Comment by downrealbad** [comment]
Agree with @taiyi747 here. It would be really sad if these smaller sites died as a result of this.

Maybe @xtekky can find a way to balance accessibility of APIs with capital power of selected providers.
